ping/rust/src/bin/testplan_0500.rs

```diff
@@ -37,18 +78,23 @@ struct OrphanRuleWorkaround(Swarm<Behaviour>);
 
 #[async_trait]
 impl PingSwarm for OrphanRuleWorkaround {
-    async fn listen_on(&mut self, address: &str) -> Result<()> {
+    async fn listen_on(&mut self, address: &str) -> Result<String> {
         let id = self.0.listen_on(address.parse()?)?;
-
         loop {
-            if let Some(SwarmEvent::NewListenAddr { listener_id, .. }) = self.0.next().await {
+            if let Some(SwarmEvent::NewListenAddr {
+                            listener_id,
+                            address,
+                        }) = self.0.next().await
+            {
                 if listener_id == id {
-                    break;
+                    debug!(
+                        "NewListenAddr event: listener_id={:?}, address={:?}",
+                        &listener_id, &address
+                    );
```

Comment:
Unless you deem the listener ID very important, we could move this log to the ping swarm so all tests benefit from it.